qiita_db/analysis.py

```diff
@@ -812,7 +812,7 @@ def build_files(self,
             # doing it here cause it's computational cheaper
             all_ids = list(chain.from_iterable([
                 samps for _, samps in viewitems(samples)]))
-            rename_dup_samples = ((len(all_ids) != len(set(all_ids))) or
+            rename_dup_samples = ((len(all_ids) != len(set(all_ids))) and
```

Comment:
Shouldn't a test be added to this specific change gets caught? The fact that the code got merged earlier points out that the unit tests are not enough to catch this issue?